test/uk/gov/hmrc/platopsgithubproxy/connector/GitHubConnectorSpec.scala

```diff
@@ -0,0 +1,150 @@
+      githubConfig = new GitHubConfig(Configuration(
+        "github.rest.api.url"    -> wireMockUrl,
+        "github.open.api.rawurl" -> wireMockUrl,
+        "github.open.api.token"  -> testToken
+      ))
+    )
+
+  implicit val headerCarrier: HeaderCarrier = HeaderCarrier()
+
+  "getGithubRawContent" should {
+
+    "return 200 with response body" in {
+      stubFor(
+        get(urlEqualTo("/hmrc/service-one/test?query=test"))
+          .willReturn(aResponse().withStatus(200).withBody("Hello World"))
+      )
+
+      val response = githubConnector
+        .getGithubRawContent("service-one", "test", Map("query" -> Seq("test")))
+        .futureValue
+
+      response.isRight shouldBe true
+      response.map(_.body) shouldBe Right("Hello World")
+      response.map(_.status) shouldBe Right(200)
+    }
+
+    "return 404 when url not found in GitHub" in {
+      stubFor(
+        get(urlEqualTo("/hmrc/service-one/test?query=test"))
+          .willReturn(aResponse().withStatus(404))
+      )
+
+      val response = githubConnector
+        .getGithubRawContent("service-one", "test", Map("query" -> Seq("test")))
+        .futureValue
+
+      response.isLeft shouldBe true
+    }
+  }
+
+  "getGithubRestContent" should {
+
+    "return 200 with response body" in {
+      stubFor(
+        get(urlEqualTo("/repos/hmrc/service-one/test?query=test"))
+          .willReturn(aResponse().withStatus(200).withBody("Hello World"))
+      )
+
+      val response = githubConnector
+        .getGithubRestContent("service-one", "test", Map("query" -> Seq("test")))
+        .futureValue
+
+      response.isRight shouldBe true
+      response.map(_.body) shouldBe Right("Hello World")
+      response.map(_.status) shouldBe Right(200)
+    }
+
+    "return 404 when url not found in GitHub" in {
+      stubFor(
+        get(urlEqualTo("/repos/hmrc/service-one/test?query=test"))
+          .willReturn(aResponse().withStatus(404))
+      )
+
+      val response = githubConnector
+        .getGithubRestContent("service-one", "test", Map("query" -> Seq("test")))
+        .futureValue
+
+      response.isLeft shouldBe true
+    }
+
+    "return 401 when unauthenticated" in {
+      stubFor(
+        get(urlEqualTo("/repos/hmrc/service-one/test?query=test"))
+          .willReturn(aResponse().withStatus(401))
+      )
+
+      val response = githubConnector
+        .getGithubRestContent("service-one", "test", Map("query" -> Seq("test")))
+        .futureValue
+
+      response.isLeft shouldBe true
+    }
+  }
+
+  "extractQueryParams" should {
+
+    "extract params with single key and value" in {
+      val queryParams = Map("pulls" -> Seq("open"))
+      GitHubConnector.extractQueryParams(queryParams) shouldBe Map("pulls" -> "open")
+    }
+
+    "extract params with single key and multiple values" in {
+      val queryParams = Map("pulls" -> Seq("open", "closed"))
+      GitHubConnector.extractQueryParams(queryParams) shouldBe Map("pulls" -> "open,closed")
```

Comment:
the concern with escaping of `,` is once embedded in the `url` interpolator.
I don't think we need to test `extractQueryParams` directly - that's just implementation details of the connector tests above.